sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/Models/DataFeedSourcesTests.cs

```diff
@@ -230,44 +232,44 @@ private void UpdateSecret(DataFeedSource dataSource, string secretPropertyName)
             switch (dataSource)
             {
                 case AzureApplicationInsightsDataFeedSource d:
-                    d.UpdateApiKey("new_secret");
+                    d.UpdateApiKey(NewSecret);
```

Comment:
It seems that Cred Scan doesn't like the name "ApiKey" followed by a string literal, so I'm using a constant instead to remove the false positive.

Reply:
I am sure missing something. Does this change the recordings? or how is credscan detecting it and how with this change it is no longer doing so?

Reply:
This is not a live test, so there's no recording to change.

The error message we get back from CredScan is:

> "A potential secret was detected in 'DataFeedSourcesTests.cs':(CSCAN-GENERAL0130 Client Secret / Api Key) Validate file contains secrets, remove, roll credential, and use approved store. For additional information on secret remediation see https://aka.ms/credscan."

and it points to the string literal `"new_secret"` as a leaked secret (false positive) in:

```cs
d.UpdateApiKey("new_secret");
```

While it doesn't do the same for other lines where `"new_secret"` appears.

From my understanding, it seems CredScan thinks the string literal that follows the keyword "ApiKey" could be a potential leak (which is a reasonable guess). If instead of passing the string literal and pass a constant:
```cs
const string NewSecret = "new_secret";
d.UpdateApiKey(NewSecret);
```
it knows `UpdateApiKey(NewSecret)` does not leak any secret because we're not passing a string literal anymore, just a symbol name. It also does not detect any secret-like string in the const declaration.